crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs

```diff
@@ -104,8 +104,7 @@ impl Simulator {
         capacity: u64,
         max_block_bytes: u64,
         fullness_and_bytes: Vec<(u64, u64)>,
-        // blocks: Enumerate<Zip<Iter<(u64, u64)>, Iter<Option<Vec<RecordedBlock>>>>>,
-        blocks: impl Iterator<Item = (usize, ((u64, u64), &'a Option<Vec<RecordedBlock>>))>,
+        blocks: impl Iterator<Item = (usize, ((u64, u64), &'a Option<(Range<u32>, u128)>))>,
```

Comment:
nit: this type seems complicated enough that a named type should make it more readable

Reply:
Done.